### PR TITLE
Deemphasize Redpanda Keeper in rpk docs

### LINKED ISSUE #1773: Deemphasize "Redpanda Keeper" in rpk docs
### Issue description

In help-docs:

> The rpk introduction prominently refers to rpk as “Redpanda Keeper” https://docs.redpanda.com/docs/beta/reference/rpk/rpk-commands/
but we basically never refer to rpk as Redpanda Keeper anywhere and never mention that.

### Updates to existing documentation

- Move "Redpanda Keeper" from first paragraph on https://docs.redpanda.com/docs/beta/reference/rpk/rpk-commands/
- Review all docs and deemphasize "Redpanda Keeper" as needed

### Link to Redpanda Slack conversation

https://redpandadata.slack.com/archives/C01LK3UK2K1/p1689773108672319

### PR DESCRIPTION
**Pull request by shorgi**
Fixes #1773 

Removes all mentions of Redpanda Keeper, except for a note in the rpk command reference.

Preview:
- Edited [rpk command reference](https://deploy-preview-1774--redpanda-documentation.netlify.app/docs/reference/rpk/rpk-commands/) with note about Redpanda Keeper

### CODE REVIEW

File: docs/reference/rpk/rpk-commands.mdx [hunk 1]
```diff
@@ -8,12 +8,16 @@ title: rpk
...
 </head>
 
-Redpanda Keeper `rpk` is Redpanda's command line interface (CLI) utility.
+The `rpk` application is Redpanda's command line interface (CLI) utility.
 
 The `rpk` commands let you configure, manage, and tune Redpanda clusters. They also let you manage topics, groups, and access control lists (ACLs).
 
 The `rpk` commands feature autocompletion, so you that can write the first few letters that uniquely identify a command and then use the tab key to complete it. For more information, see [rpk autocompletion documentation](../../rpk/rpk-generate/rpk-generate-shell-completion).
 
+:::note
```
Comment (twmb) [inline docs/reference/rpk/rpk-commands.mdx:17]: maybe at the bottom rather than top still? And maybe mention it's trivia?

Comment (Feediver1) [reply docs/reference/rpk/rpk-commands.mdx:17]: @twmb We would not make mention of any functionality as trivia. If you don't believe this is useful/helpful information for users, please let us know. We can move content--no problem.

File: docs/reference/rpk/rpk-commands.mdx [hunk 2]
```diff
@@ -8,12 +8,16 @@ title: rpk
...
+The name `rpk` stands for Redpanda Keeper.
```
Comment (Feediver1) [inline docs/reference/rpk/rpk-commands.mdx:18]: What is the value-add for customers to keep this in given the guidance/request to remove it entirely?

File: versioned_docs/version-21.11/reference/rpk-commands.mdx [hunk 3]
```diff
@@ -8,10 +8,14 @@ title: rpk commands
   <link rel="canonical" href="https://docs.redpanda.com/docs/reference/rpk/rpk-commands/" />
 </head>
 
-`rpk` (Redpanda Keeper) is Redpanda's command line interface (CLI) utility. `rpk` commands allow you to configure and manage Redpanda clusters, tune them for better performance, manage topics and groups, manage access control lists (ACLs).
+`rpk` is Redpanda's command line interface (CLI) utility. `rpk` commands allow you to configure and manage Redpanda clusters, tune them for better performance, manage topics and groups, manage access control lists (ACLs).
 
 This section lists each `rpk` command in alphabetical order, along with a table of flags for that command. All descriptions are from the output of the `rpk <command> –help` command.
 
+:::note
+The name `rpk` stands for Redpanda Keeper.
```
Comment (Feediver1) [inline versioned_docs/version-21.11/reference/rpk-commands.mdx:16]: see comment above re this

**Review by Feediver1 — APPROVED** [review]
Please remove Redpanda Keeper note, as it does not add any real value for our customers, and the original request is to remove it altogether. Thx.

File: docs/reference/rpk/rpk-commands.mdx (hunk 2 above)
Comment (shorgi) [reply docs/reference/rpk/rpk-commands.mdx:18]: @Feediver1 - the original request from @twmb in Slack was to move `Redpanda Keeper` from the beginning of the first paragraph to somewhere further down, perhaps presented as a nice piece of trivia. 

The value for the customer is insight about origin of the name `rpk`.

File: docs/reference/rpk/rpk-commands.mdx (hunk 1 above)
Comment (shorgi) [reply docs/reference/rpk/rpk-commands.mdx:17]: @twmb - two things:
1. This is the furthest down the page for content about `rpk` itself. The rest is the rpk help's text. Alternatively, would moving this note to the bottom of https://docs.redpanda.com/docs/reference/rpk/ (after the complete list of commands) work better for you?
2. The heading of the callout (currently `Note`) can be customized. Changing it to 'Trivia' or 'Factoid' would be more accurate. We don't have a precedent for these callouts, however, from our style guide, so best to leave this a Note in this PR, then change it after internal discussion.

Comment (micheleRP) [reply docs/reference/rpk/rpk-commands.mdx:17]: @shorgi note the typo in the intro:  "so you that can"

Comment (shorgi) [reply docs/reference/rpk/rpk-commands.mdx:17]: Decision: keep the Note here (rpk-commands.mdx), and fix the typo

**Pull request merged**